packages/apiserver/src/resources/materials/material.interceptor.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  CallHandler,
+  ExecutionContext,
+  Injectable,
+  NestInterceptor,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Material } from './material.entity';
+import { DigitalOceanService } from '@mimir/api-util';
+import { FileService } from '../../file/file.service';
+import { readLocalFile } from '@mimir/helper-functions';
+
+@Injectable()
+export class TransformInterceptor<T> implements NestInterceptor<T, Material[]> {
+  constructor(
+    private digitalOceanService: DigitalOceanService,
+    private fileService: FileService
+  ) {}
+
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler
+  ): Observable<Material[]> {
+    return next.handle().pipe(
+      map((items: Material[]) =>
+        items.map((item) => {
+          if (item.picture.includes('localhost')) {
+            const file = readLocalFile(item.picture);
+            if (file.buffer) {
+              this.digitalOceanService
+                .createFile({
+                  originalname: file.originalname,
+                  buffer: file.buffer,
+                })
+                .then((newName) => {
+                  Material.update(item.id, { picture: newName.toString() });
+                });
+              setTimeout(() => {
```

Comment:
Suggestion: Let's remove `setTimeout` and do this callback inside one more `then` block:

```typescript
this.digitalOceanService.createFile({
   originalname: file.originalname,
   buffer: file.buffer,
})
.then(newName) => Material.update(item.id, { picture: newName.toString() }) )
.then(() => {
  this.fileService.removeFile(item.picture.split('/').pop());
});
```